templates/documents/upload_files_modal.html

```diff
@@ -0,0 +1,74 @@
+{% extends 'documents/shared/modal_body.html' %}
+
+{% block modal_body %}
+    <div class="mt-2 flex flex-col gap-5">
+        <input
+            x-ref="fileUpload"
+            name="files"
+            @change="handleFileChange($event)"
+            type="file"
+            multiple
+            class="file-input file-input-bordered w-full max-w-xs"
+            accept=".txt,.pdf,.doc,.docx,.xls,.xlsx,.csv,.jpg,.jpeg,.png,.gif,.bmp,.webp,.svg,.mp4,.mov,.avi,.mp3,.wav"/>
+
+        <select class="select select-bordered w-full" name="collection_name">
+            {% for collection in collections %}
+                <option value="{{ collection.name }}">{{ collection.name }}</option>
+            {% endfor %}
+        </select>
+
+        <template x-for="(file, index) in uploaded_files" :key="index">
+            <div class="grid grid-cols-2 gap-1 border-b border-base-300 py-3 items-center">
+                <span class="col-span-1" x-text="file.name"></span>
+                <div class="col-span-1 flex flex-row gap-1 items-center">
+                    <textarea
+                        x-model="file_summaries[file.name]"
+                        class="col-span-1 textarea textarea-bordered w-full placeholder:text-gray-500"
+                        maxlength="{{ max_summary_length }}"
+                        placeholder="Enter a summary of the file..."
+                    ></textarea>
+                    <button aria-label="Remove File" title="Remove File" type="button" class="btn btn-xs" @click="removeFile(index)">
+                        <i class="fa-solid fa-trash htmx-hide"></i>
+                    </button>
+                </div>
+            </div>
+        </template>
+        <input type="hidden" name="file_summaries" :value="JSON.stringify(file_summaries)" />
```

Comment:
just curious why you decided to use JSON here instead of standard form fields which already being displayed.

Reply:
I simplified it [here](https://github.com/dimagi/open-chat-studio/pull/1259/commits/d2f5a2af24fd0df17c3b3c3315d00c5998fffc31). I did not think of this way back